test/test_cachefile.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+import hashlib
+from KTLibrary.KTLibrary import KTLibrary
+
+# Function that caculate the hash of a given file
+def calculate_file_hash(path):
+    hasher = hashlib.sha256()
+    with open(path, 'rb') as f:
+        for chunk in iter(lambda: f.read(4096), b''):
+            hasher.update(chunk)
+    return hasher.hexdigest()
+
+def test_read_cache_file():
+    testLibrary = KTLibrary()
+    file_path = 'SampleCacheFiles/convolution_A100.json'
+
+    testLibrary.read_file(file_path)
+
+    # Read device name of the given file
+    file = testLibrary.cache_file
+    device_name = file.get('device_name')
+   
+    # Check if the expected value of device name is in the file
+    assert device_name == 'NVIDIA A100-PCIE-40GB'
+
+def test_write_cache_file():
+    testLibrary = KTLibrary()
+    file_path = 'SampleCacheFiles/convolution_A100.json'
+
+    testLibrary.read_file(file_path)
+    p = Path.joinpath(Path(os.getcwd()), 'KTLibrary', 'test.json')
```

Comment:
In het algemeen kan je geen aannames maken over van waar de test file gerunt wordt, dus maak de paths file-relative met gebruikt van ``__file__``.